Public/Src/Engine/Scheduler/PipExecutor.cs

```diff
@@ -2851,21 +2851,12 @@ private static int GetCacheLevelForLocality(PublishedEntryRefLocality locality)
                 new List<(FileArtifact, FileMaterializationInfo)>(pip.Outputs.Length);
 
             // Outputs should be the same as what was in the metadata section.
-            int outputHashIndex = 0;
-            for (int i = 0; i < pip.Outputs.Length; i++)
+            for (int i = 0; i < metadata.StaticOutputHashes.Count; i++)
             {
-                FileArtifactWithAttributes attributedOutput = pip.Outputs[i];
-                if (!attributedOutput.CanBeReferencedOrCached())
-                {
-                    // Skipping non-cacheable outputs (note that StoreContentForProcess does the same).
-                    continue;
-                }
-

```

Comment:
Why is the skip output not preserved in the new code?

Reply:
Since I changed the for loop. Instead of looping outputs, I loop the metadata.StaticOutputHashes, which are all cacheable outputs. so I don't need to preserver this skip.